tests/tensorflow/test_keras_pyfunc_model_works_with_all_input_types.py

```diff
@@ -48,22 +63,99 @@ def multi_tensor_input_model(data):
     model = Model(inputs=[input_a, input_b], outputs=output)
     model.compile(loss="mean_squared_error", optimizer=SGD())
     model.fit([x.values[:, :2], x.values[:, -2:]], y)
-    return model
+
+    signature = ModelSignature(
+        inputs=Schema(
+            [
+                TensorSpec(np.dtype(np.float64), (-1, 2), "a"),
+                TensorSpec(np.dtype(np.float64), (-1, 2), "b"),
+            ]
+        )
+    )
+    return model, signature
+
+
+@pytest.fixture(scope="module")
+def single_multidim_tensor_input_model(data):
+    """
+    This is a model that requires a single input of shape (-1, 4, 3)
+    """
+    x, y = data
+    model = Sequential()
+
+    def f1(z):
+        from tensorflow.keras import backend as K
+
+        return K.mean(z, axis=2)
+
+    model.add(Lambda(f1))
+    model.add(Dense(3, input_dim=4))
+    model.add(Dense(1))
+    model.compile(loss="mean_squared_error", optimizer=SGD())
+    model.fit(np.repeat(x.values[:, :, np.newaxis], 3, axis=2), y.values)
+    signature = ModelSignature(
+        inputs=Schema(
+            [
+                TensorSpec(np.dtype(np.float64), (-1, 4, 3)),
+            ]
+        )
+    )
+    return model, signature
+
+
+@pytest.fixture(scope="module")
+def multi_multidim_tensor_input_model(data):
+    """
+    This is a model that requires 2 inputs: 'a' and 'b',
+    input 'a' must be shape of (-1, 2, 3),
+    input 'b' must be shape of (-1, 2, 5),
+    """
+    x, y = data
+    input_a = Input(shape=(2, 3), name="a")
+    input_b = Input(shape=(2, 5), name="b")
+
+    def f1(z):
+        from tensorflow.keras import backend as K
+
+        return K.mean(z, axis=2)
+
+    input_a_sum = Lambda(f1)(input_a)
+    input_b_sum = Lambda(f1)(input_b)
```

Comment:
```suggestion
    input_a_sum = Lambda(lambda z: K.mean(z, axis=2))(input_a)
    input_b_sum = Lambda(lambda z: K.mean(z, axis=2))(input_b)
```

Can we use lambda?

Reply:
We have to add `from tensorflow.keras import backend as K` inside the lambda function, otherwise when the model is pickled and unpickled, it does not work.

Reply:
Got it.